code/game/machinery/engi_points_vendor.dm

```diff
@@ -0,0 +1,689 @@
+	if(safety || !bomb_set)
+		timing = FALSE
+		return
+	exploding = TRUE
+	update_icon()
+	for(var/mob/M in player_list)
+		send_to_playing_players('sound/machines/Alarm.ogg')
+	var/turf/startpoint = get_turf(src)
+	sleep(100)
+	for(var/mob/M in player_list)
+		send_to_playing_players('sound/effects/explosionfar.ogg')
+	spawn_atom_to_turf(/obj/effect/overlay/temp/explosion, startpoint, 1, FALSE)
+	qdel(src)
+	for(var/I in spiral_range_turfs(range, startpoint))
+		var/turf/T = I
+		if(!T)
+			continue
+		if(istype(T, /turf/open/floor))
+			T.ChangeTurf(payload_floor)
+			spawn_atom_to_turf(/obj/effect/overlay/temp/fire, T, 1, FALSE)
+		else if(istype(T, /turf/closed/wall))
+			T.ChangeTurf(payload_wall)
+			spawn_atom_to_turf(/obj/effect/overlay/temp/fire, T, 1, FALSE)
+		CHECK_TICK
+
+
+
+// DISCO BALL
+
+
+
+/obj/machinery/disco
+	name = "Radiant Dance Machine Mark IV"
+	desc = "The first three prototypes were discontinued after mass casualty incidents."
+	icon = 'icons/obj/lighting.dmi'
+	icon_state = "disco0"
+	anchored = FALSE
+	verb_say = "states"
+	density = TRUE
+	req_access = list(access_engine)
+	var/active = FALSE
+	var/list/rangers = list()
+	var/list/listeners = list()
+	var/charge = 35
+	var/stop = 0
+	var/list/available = list()
+	var/list/select_name = list()
+	var/list/spotlights = list()
+	var/list/sparkles = list()
+	var/static/list/songs = list(
+		new /datum/track("Engineering's Basic Beat", 					'sound/misc/disco.ogg', 	600, 	5),
+		new /datum/track("Engineering's Domination Dance", 				'sound/misc/e1m1.ogg', 		950, 	6),
+		new /datum/track("Engineering's Superiority Shimmy", 			'sound/misc/Paradox.ogg', 	2400, 	4),
+		new /datum/track("Engineering's Ultimate High-Energy Hustle",	'sound/misc/boogie2.ogg',	1770, 	5),
+		)
+	var/datum/track/selection = null
+
+/datum/track
+	var/song_name = "generic"
+	var/song_path = null
+	var/song_length = 0
+	var/song_beat = 0
+
+/datum/track/New(name, path, length, beat)
+	song_name = name
+	song_path = path
+	song_length = length
+	song_beat = beat
+
+/obj/machinery/disco/Initialize()
+	..()
+	selection = songs[1]
+
+
+/obj/machinery/disco/Destroy()
+	dance_over()
+	return ..()
+
+/obj/machinery/disco/attackby(obj/item/O, mob/user, params)
+	if(!active)
+		if(istype(O, /obj/item/weapon/wrench))
+			if(!anchored && !isinspace())
+				to_chat(user,"<span class='notice'>You secure the [src] to the floor.</span>")
+				anchored = TRUE
+			else if(anchored)
+				to_chat(user,"<span class='notice'>You unsecure and disconnect the [src].</span>")
+				anchored = FALSE
+			playsound(src, 'sound/items/Deconstruct.ogg', 50, 1)
+			return
+	return ..()
+
+
+/obj/machinery/disco/interact(mob/user)
+	if (!anchored)
+		to_chat(user,"<span class='warning'>This device must be anchored by a wrench!</span>")
+		return
+	if(!allowed(user))
+		to_chat(user,"<span class='warning'>You are overwhelmed by the raw amount of data being displayed, only an engineer could operate such a sophisticated device.</span>")
```

Comment:
Change this message, `allowed` only checks the ID privileges of the user and this talks about `knowledge`.